File: clean-now-api/src/main/java/com/cleannoww/cleannowapi/domain/Cliente.java
package com.cleannoww.cleannowapi.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import com.fasterxml.jackson.annotation.JsonBackReference;
import java.util.ArrayList;
import java.util.List;
@Entity
@Table(name="Clientes")
@Getter
@Setter
@NoArgsConstructor
public class Cliente {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="id_cliente")
    private int id_cliente;
    @Column(name="dni")
    private int dni;
    @Column(name="nombre_completo")
    private String nombre_completo;
    @Column(name="email")
    private String email;
    @Column(name="contraseña")
    private String contraseña;
    @Column(name="tipoRegistro")
    private String tipoRegistro;
    @Column(name="direccion")
    private String direccion;
    @OneToMany(mappedBy = "cliente")
    private List<ReservaServicio> historialReservas;
    @OneToMany(mappedBy = "cliente")
    private List<Evaluacion> historialEvaluaciones;



}

File: cleanangular/src/app/model/cliente.ts
import { Evaluacion } from "./evaluacion";
import { ReservaServicio } from "./reservaservicio";

export interface Cliente{
    id_cliente:number;
    dni:number;
    nombre_completo:String;
    email:String;
    contraseña:String;
    tipoRegistro:String;
    direccion:String;
    historialReservas:ReservaServicio[];
    historialEvaluaciones:Evaluacion[];
}